streams/src/test/java/org/apache/kafka/streams/state/internals/WindowStoreBuilderTest.java

```diff
@@ -164,17 +160,6 @@ public void shouldThrowNullPointerIfInnerIsNull() {
         assertThrows(NullPointerException.class, () -> new WindowStoreBuilder<>(null, Serdes.String(), Serdes.String(), new MockTime()));
     }
 
-    @Test
-    public void shouldThrowNullPointerIfKeySerdeIsNull() {
-        assertThrows(NullPointerException.class, () -> new WindowStoreBuilder<>(supplier, null, Serdes.String(), new MockTime()));
-    }
-
-    @Test
-    public void shouldThrowNullPointerIfValueSerdeIsNull() {
-        assertThrows(NullPointerException.class, () -> new WindowStoreBuilder<>(supplier, Serdes.String(),
-            null, new MockTime()));
-    }
```

Comment:
These tests were buggy - there aren't currently any `null` checks in place for key / value serdes. The only reason they were passing earlier is due to the use of `EasyMock` with ["nice" mocks](https://easymock.org/user-guide.html#mocking-nice). 

In the `setUp` method, the expectation for the `WindowBytesStoreSupplier::name` method is setup - https://github.com/apache/kafka/blob/7782741262c08e5735f7c8e09727ec37cb5f7f02/streams/src/test/java/org/apache/kafka/streams/state/internals/WindowStoreBuilderTest.java#L61

 By default, this method stub covers only the first invocation of the method and subsequent invocations will result in a `null` value being returned (because the mock object is "nice"). The first invocation occurs in the `setUp` method itself when the builder is instantiated - https://github.com/apache/kafka/blob/7782741262c08e5735f7c8e09727ec37cb5f7f02/streams/src/test/java/org/apache/kafka/streams/state/internals/WindowStoreBuilderTest.java#L65-L70
https://github.com/apache/kafka/blob/7782741262c08e5735f7c8e09727ec37cb5f7f02/streams/src/main/java/org/apache/kafka/streams/state/internals/WindowStoreBuilder.java#L39

The `null` check for name here - https://github.com/apache/kafka/blob/7782741262c08e5735f7c8e09727ec37cb5f7f02/streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractStoreBuilder.java#L41 
is what resulted in the `NullPointerException` being thrown in the `shouldThrowNullPointerIfKeySerdeIsNull` and `shouldThrowNullPointerIfValueSerdeIsNull` tests.

This test bug was discovered because method stubbings in `Mockito` are applied to any number of invocations of the method by default.

Reply:
I checked the other *StoreBuilder files and none of them have a check for a null on the keySerde or the valueSerde, so I am okay with these tests being removed. I believe that if we do choose to remove the tests we should add a Jira task to correct all *StoreBuilders to have such a null check since other more trivial fields do have such checks.

Reply:
There were similar buggy tests that were removed in the `SessionStoreBuilderTest` migration as well - https://github.com/apache/kafka/pull/14142#discussion_r1287021377. I think not having a `null` check on the key / value serdes is intentional - a default key / value serdes is picked up from the processor context later on if the key / value serdes in the builder is `null`.